### DISCUSSION TITLE
How to invoke an operation using reflective api

### DISCUSSION THREAD

**Discussion by nyemiks** [openmdx/openmdx#29] [Q&A] [answered]
How can I invoke an mdx operation using the reflective api ?
I know how to do this using the typed jmi api.

I would appreciate code snippets or (if available) reference to an
existing example.

**Comment by wfro64** [comment, marked as answer]
Every object of the JMI API inherits from javax.jmi.reflect.RefFeatured which defines the operation refInvokeOperation: https://github.com/openmdx/openmdx/blob/master/core/src/java/javax/jmi/reflect/RefFeatured.java

     public Object refInvokeOperation(String requestedOperation, List args) throws RefException;

Examples:

- https://github.com/openmdx/openmdx/blob/master/core/src/java/org/openmdx/application/rest/adapter/InboundConnection_2.java
- https://github.com/openmdx/openmdx/blob/master/portal/src/java/org/openmdx/portal/servlet/action/InvokeOperationAction.java
- https://github.com/openmdx/openmdx/blob/master/test-core/src/java/test/mock/clock1/Segments.java

**Reply by nyemiks** [reply]
Thank you. I will go through this sources and revert
